tuplex/core/src/physical/TransformTask.cc

```diff
@@ -621,6 +616,34 @@ namespace tuplex {
 
         // go over all input partitions.
         for(const auto &inputPartition : _inputPartitions) {
+            size_t numTopCompleted = 0;
+            size_t numBottomCompleted = 0;
+            bool isTopLimitReached = false;
+            bool isBottomLimitReached = false;
+
+            tuplex::g_rowsDoneMutex.lock();
```

Comment:
this logic here looks like it should go into a function. limitsReached?

Reply:
could also potentially short-circuit:

I.e., if no bottom limit is specified for a task -> bottom limit is reached.